openapiGenerator.go

```diff
@@ -116,29 +98,61 @@ type openapiGenerator struct {
 	// @solo.io customization to support enum validation schemas with int or string values
 	// we need to support this since some controllers marshal enums as integers and others as strings
 	enumAsIntOrString bool
+
+	// @solo.io customizations to define schemas for certain messages
+	customSchemasByMessageName map[string]openapi3.Schema
 }
 
 type DescriptionConfiguration struct {
 	// Whether or not to include a description in the generated open api schema
 	IncludeDescriptionInSchema bool
-
-	// The maximum number of characters to include in a description
-	// If IncludeDescriptionsInSchema is set to false, this will be ignored
-	// A 0 value will be interpreted as "include all characters"
-	// Default: 0
-	MaxDescriptionCharacters int
 }
 
-func newOpenAPIGenerator(model *protomodel.Model, perFile bool, singleFile bool, yaml bool, useRef bool, descriptionConfiguration *DescriptionConfiguration, enumAsIntOrString bool) *openapiGenerator {
+func newOpenAPIGenerator(
+	model *protomodel.Model,
+	perFile bool,
+	singleFile bool,
+	yaml bool,
+	useRef bool,
+	descriptionConfiguration *DescriptionConfiguration,
+	enumAsIntOrString bool,
+	messagesWithEmptySchema []string,
+) *openapiGenerator {
 	return &openapiGenerator{
-		model:                    model,
-		perFile:                  perFile,
-		singleFile:               singleFile,
-		yaml:                     yaml,
-		useRef:                   useRef,
-		descriptionConfiguration: descriptionConfiguration,
-		enumAsIntOrString:        enumAsIntOrString,
+		model:                      model,
+		perFile:                    perFile,
+		singleFile:                 singleFile,
+		yaml:                       yaml,
+		useRef:                     useRef,
+		descriptionConfiguration:   descriptionConfiguration,
+		enumAsIntOrString:          enumAsIntOrString,
+		customSchemasByMessageName: buildCustomSchemasByMessageName(messagesWithEmptySchema),
+	}
+}
+
+func buildCustomSchemasByMessageName(messagesWithEmptySchema []string) map[string]openapi3.Schema {
```

Comment:
For ease of maintenance how would you feel about adding a comment here.
Perhaps a short form of that which is above the specialsolotypes

Reply:
Yeah great call, will do